src/commands/set-default.ts

```diff
@@ -74,15 +53,92 @@ export default class SetDefault extends APICommand {
       console.warn("Note: --confidential is implied when using --secret, so you don't need to specify both.")
     }
 
-    const config = reforge.raw(key)
+    // Validate required arguments before making API calls
+    if (!args.name && !isInteractive(flags)) {
+      return this.err("'name' argument is required when interactive mode isn't available.")
+    }
+
+    if (!flags.environment && !isInteractive(flags)) {
+      return this.err("'environment' is required when interactive mode isn't available.")
+    }
+
+    // Fetch all configs from metadata endpoint
+    const metadataRequest = await this.apiClient.get('/all-config-types/v1/metadata')
+
+    if (!metadataRequest.ok) {
+      const errorMsg = metadataRequest.error?.error || `Failed to fetch configs: ${metadataRequest.status}`
+      return this.err(errorMsg, {serverError: metadataRequest.error})
+    }
+
+    interface ConfigMetadata {
```

Comment:
Note to self: This is already the 3rd time I've seen this interface defined in a different file